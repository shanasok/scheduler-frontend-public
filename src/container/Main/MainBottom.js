import React from 'react';

export class MainBottom extends React.Component {

    render() {
        return <div id="mainBottom">
            Main Bottom goes here
        </div>
    }
}