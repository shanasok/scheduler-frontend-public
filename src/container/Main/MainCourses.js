import React from 'react';
import {CourseOption} from "../CourseOption/CourseOption";

export function populateCourseOption(courses){
    if (courses && Array.isArray(courses)){
        return courses.map(course =>
            <CourseOption key={course.id} course={course}/>
        )
    }
}


export default class MainCourses extends React.Component {

    render() {
        const courseOptions = populateCourseOption(this.props.courses);

        return <div id="mainCourses">
            {courseOptions}
        </div>
    }
}