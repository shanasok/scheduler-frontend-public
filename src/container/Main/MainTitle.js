import React from 'react';

export class MainTitle extends React.Component {

    render() {
        return <div id="mainTitle">
            Click to see course content, materials and scheduling information.
        </div>
    }
}