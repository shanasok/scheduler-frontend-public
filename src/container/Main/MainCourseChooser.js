import React from 'react';
import { connect } from "react-redux";

import {MainBottom} from "./MainBottom";
import MainCourses from "./MainCourses";
import {MainTitle} from "./MainTitle";

import '../../css/CourseOptionStyles.css';

const MainCourseChooser = props => (
    <div id="mainCourseChooser">
        <MainTitle />
        <MainCourses courses={props.courses} />
        <MainBottom />
    </div>
);

const mapStateToProps = state => ({
    courses: state.courses,
});

export default connect(mapStateToProps)(MainCourseChooser)