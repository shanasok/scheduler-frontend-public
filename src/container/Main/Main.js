import React from 'react';
import {Switch, Route, BrowserRouter as Router} from "react-router-dom";
import MainCourseChooser from "./MainCourseChooser";
import CourseDetails from "../courseBreakdown/CourseDetails"
import LessonSlide from "../courseBreakdown/subjectBreakdown/lessonBreakdown/LessonSlide";
import NotFound from "../NotFound";

export class Main extends React.Component {

    render() {
        return <div id="main">
            <Router>
                <Switch>
                    <Route exact path="/" component={() => <MainCourseChooser />} />
                    <Route exact path="/course" component={() => <CourseDetails />} />
                    <Route exact path="/lesson" component={() => <LessonSlide />} />
                    <Route component={() => <NotFound/>} />
                </Switch>
            </Router>
        </div>
    }
}