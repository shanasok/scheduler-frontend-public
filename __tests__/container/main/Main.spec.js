import React from 'react';
import {Main} from '../../../src/container/Main/Main';
import renderer from 'react-test-renderer';
import {Provider as ReduxProvider} from "react-redux";
import configureMockStore from "redux-mock-store";
import Adapter from 'enzyme-adapter-react-16';
import {mount, configure, shallow} from "enzyme";
import {MemoryRouter} from "react-router";
import MainCourseChooser from "../../../src/container/Main/MainCourseChooser";

require('console.table');

const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: []
}

const course =  {
    name: "testCourseName",
        summary: "testCourseSummary",
        subjects: [
        {
            name: "testSubjectName",
            summary: "testSubjectSummary",
            subjectNumber: 1,
            lessons: [
                lesson
            ],
        }
    ]
}

const mockStore = configureMockStore()
const mockState = {
    lesson: lesson
}
let store = mockStore(mockState)

it('renders correctly', () => {
    const tree = renderer
        .create(<ReduxProvider store={store}><Main/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

configure({adapter: new Adapter()});
it('valid path mainCourseChooser', () => {
    let wrapper = mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <ReduxProvider store={store}>
                <Main/>
            </ReduxProvider>
        </MemoryRouter>
    );
    expect(wrapper.find(MainCourseChooser)).toHaveLength(1);
    wrapper.unmount();
});

// it('invalid path retrieves not found', () => {
//     let wrapper = shallow(
//         <MemoryRouter initialEntries={[ '/unknown' ]} >
//             <ReduxProvider store={store}>
//                 <Main/>
//             </ReduxProvider>
//         </MemoryRouter>
//     );
//     expect(wrapper.find(NotFound)).toHaveLength(1);
// });

// it('valid path lesson', () => {
//     //        return <Link to={{ pathname: "/lesson",  state:{ lesson: this.props.lesson } }}>
//     const testProps = {
//        location: {
//             state: {
//                 lesson: lesson,
//             }
//         }
//     };
//     let wrapper = mount(
//         <MemoryRouter initialEntries={[ '/lesson' ]}>
//             <ReduxProvider store={store}>
//                 <Main {...testProps}/>
//             </ReduxProvider>
//         </MemoryRouter>
//     );
//     expect(wrapper).toBeDefined();
//     expect(wrapper.find(LessonSlide)).toHaveLength(1);
//     wrapper.unmount();
// });
