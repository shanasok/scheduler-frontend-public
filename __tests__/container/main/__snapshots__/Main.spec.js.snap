// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="main"
>
  <div
    id="mainCourseChooser"
  >
    <div
      id="mainTitle"
    >
      Click to see course content, materials and scheduling information.
    </div>
    <div
      id="mainCourses"
    />
    <div
      id="mainBottom"
    >
      Main Bottom goes here
    </div>
  </div>
</div>
`;
